src/rrdp.rs

```diff
@@ -38,36 +47,95 @@ use crate::xml::decode::{Content, Error as XmlError, Reader, Name};
 /// This type represents the decoded content of the RRDP Update Notification
 /// File. It can be read from a reader via the [`parse`][Self::parse]
 /// function. All elements are accessible as attributes.
+#[derive(Clone, Debug, Eq, PartialEq)]
 pub struct NotificationFile {
     /// The identifier of the current session of the server.
     ///
     /// Delta updates can only be used if the session ID of the last processed
     /// update matches this value.
-    pub session_id: Uuid,
-
+    session_id: Uuid,
+    
     /// The serial number of the most recent update provided by the server.
     ///
     /// Serial numbers increase by one between each update.
-    pub serial: u64,
-
+    serial: u64,
+    
     /// The URI and hash value of the most recent snapshot.
     ///
     /// The snapshot contains a complete set of all data published via the
     /// repository. It can be processed using the [`ProcessSnapshot`] trait.
-    pub snapshot: UriAndHash,
-
+    snapshot: SnapshotInfo,
+    
     /// The list of available delta updates.
     ///
-    /// The first element of the vec’s items is the serial number of the
-    /// delta and the second element is the URI of the location of the delta
-    /// and its hash. Deltas can be processed using the [`ProcessDelta`]
-    /// trait.
+    /// Deltas can be processed using the [`ProcessDelta`] trait.
     ///
-    /// Note that after parsing, the list will be in the order as received
-    /// from the server. That is, it may not be ordered by serial numbers.
-    pub deltas: Vec<(u64, UriAndHash)>,
+    /// Note that after parsing, the list will be in reverse serial order,
+    /// i.e. the highest serial will appear first, and it is ensured that
+    /// there are no gaps in the sequence. Notification files which contain
+    /// gaps will be rejected by the parser as 'Malformed'.
+    deltas: Vec<DeltaInfo>,
 }
 
+// Data Access
```

Comment:
If you use triple slashes here and mark this a section with a hash, this will also show up in the documentation which is nice.

Reply:
done